CMakeLists.txt

```diff
@@ -18,16 +18,21 @@ option(COSMA_WITH_APPS "Generate the miniapp targets." ${MASTER_PROJECT})
 option(COSMA_WITH_BENCHMARKS "Generate the benchmark targets." ${MASTER_PROJECT})
 option(COSMA_WITH_INSTALL "Enable installation." ${MASTER_PROJECT})
 option(COSMA_WITH_PROFILING "Enable profiling." OFF)
+option(COSMA_WITH_FOTRAN_BINDINGS "Make Fortran bindings to COSMA." OFF) 
 
-option(COSMA_WITH_GPU "Use CUDA cuBLAS GPU back end." OFF)
-option(COSMA_WITH_MKL_BLAS "Use MKL BLAS back end." OFF)
-option(COSMA_WITH_OPENBLAS "Use OpenBLAS backend." OFF)
-option(COSMA_WITH_NETLIB_BLAS "Use Netlib BLAS backend." OFF)
+set(COSMA_BLAS_BACKEND "MKL" CACHE STRING 
+  "Blas backend. Can be MKL, MKL_SCALAPACK, OPENBLAS, NETLIB, NETLIB_SCALAPACK, CUDA or ROCM.")
```

Comment:
`NETLIB` alone might be a typo.

Reply:
This replaces the option `COSMA_WITH_NETLIB_BLAS`. What name would you suggest?

Reply:
Sorry, you are right, my mistake!

Reply:
What about `NETLIB_BLAS`?
@kabicm  Are you sure that you want an easy option to use netlib BLAS in cosma?

Reply:
I think this option might have to be split into two options, one for BLAS and one for LAPACK, e.g.:
- `COSMA_BLAS_BACKEND` (with possible values `MKL`, `OPENBLAS`, `NETLIB`) and 
- `COSMA_LAPACK` (with possible values `MKL`, `NETLIB` and `None`), 
because scalapack is not a hard dependency of COSMA. In fact, we make different libraries in `src/cosma/CMakeLists.txt`, one `cosma` (without scalapack dependency) and `cosma_pxgemm` (cosma with scalapack wrappers, thus requiring scalapack as a dependency). 

What do you think?

Reply:
It is dangerous. Someone may mix MKL BLAS and netlib ScaLAPACK (compiled for a different blas, e.g. OPENBLAS) which is source of problem depending on the linker, static/shared library and the order of kinking.

Reply:
We've discussed it just now and I'll add a separate option with checks for matching ScaLAPACK and BLAS if no GPU backend is selected.